src/Yoti.Auth/Verifications/AgeVerificationParser.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Yoti.Auth.Verifications
+{
+    internal class AgeVerificationParser
+    {
+        private Dictionary<string, AgeVerification> _ageUnderVerificationsDict;
+        private Dictionary<string, AgeVerification> _ageOverVerificationsDict;
+        private List<AgeVerification> _allVerificationsDict;
+
+        public AgeVerificationParser(BaseProfile baseProfile)
+        {
+            _ageUnderVerificationsDict = FindAgeUnderVerifications(baseProfile);
+            _ageOverVerificationsDict = FindAgeOverVerifications(baseProfile);
+
+            _allVerificationsDict = _ageUnderVerificationsDict.Values
+                .Concat(_ageOverVerificationsDict.Values).ToList();
+        }
+
+        public ReadOnlyCollection<AgeVerification> FindAllAgeVerifications()
+        {
+            return _allVerificationsDict.AsReadOnly();
+        }
+
+        public AgeVerification FindAgeUnderVerification(int age)
+        {
+            return _ageUnderVerificationsDict?.FirstOrDefault(
+                x => x.Key == string.Format("{0}:{1}", Constants.UserProfile.AgeUnderAttribute, age))
+                .Value;
+        }
+
+        public AgeVerification FindAgeOverVerification(int age)
+        {
+            return _ageOverVerificationsDict?.FirstOrDefault(
+                x => x.Key == string.Format("{0}:{1}", Constants.UserProfile.AgeOverAttribute, age))
+                .Value;
+        }
+
+        private static Dictionary<string, AgeVerification> FindAgeUnderVerifications(BaseProfile baseProfile)
```

Comment:
We can refactor this to work for both _age_under_ and _age_over_, if we pass in a value for _Constants.UserProfile.AgeOverAttribute_ (or _AgeUnderAttribte_).  Then we won't need two methods for the same task.

Reply:
SOLI**D** suggestion